x-pack/plugins/security_solution/cypress/test_plans/detection_response/prebuilt_rules/installation_and_upgrade.md

```diff
@@ -0,0 +1,580 @@
+
+#### **Scenario: User can install prebuilt rules one by one**
+
+**Coverage**: 1 e2e test with mock rules.
+
+```Gherkin
+Given no prebuilt rules are installed in Kibana
+And there are X prebuilt rules available to install
+When user opens the Add Rules page
+Then prebuilt rules available for installation should be displayed in the table
+When user installs one individual rule
+Then success message should be displayed after installation
+And the installed rule should be removed from the table
+When user navigates back to the Rule Management page
+Then user should see a CTA to install prebuilt rules
+And user should see the number of rules available to install decreased by 1
+```
+
+#### **Scenario: User can install multiple prebuilt rules selected on the page**
+
+**Coverage**: 1 e2e test with mock rules.
+
+```Gherkin
+Given no prebuilt rules are installed in Kibana
+And there are X prebuilt rules available to install
+When user opens the Add Rules page
+Then prebuilt rules available for installation should be displayed in the table
+When user selects <Y> rules
+Then user should see a CTA to install <Y> number of rules
+When user clicks the CTA
+Then success message should be displayed after installation
+And all the installed rules should be removed from the table
+When user navigates back to the Rule Management page
+Then user should see a CTA to install prebuilt rules
+And user should see the number of rules available to install decreased by <Y> number of installed rules
+
+Examples:
+  | Y                               |
+  | a few rules on the page, e.g. 2 |
+  | all rules on the page, e.g. 12  |
+```
+
+#### **Scenario: User can install all available prebuilt rules at once**
+
+**Coverage**: 1 e2e test with mock rules.
+
+```Gherkin
+Given no prebuilt rules are installed in Kibana
+And there are X prebuilt rules available to install
+When user opens the Add Rules page
+Then prebuilt rules available for installation should be displayed in the table
+When user installs all rules
+Then success message should be displayed after installation
+And all the rules should be removed from the table
+And user should see a message indicating that all available rules have been installed
+And user should see a CTA that leads to the Rule Management page
+When user clicks on the CTA
+Then user should be navigated back to Rule Management page
+And user should NOT see a CTA to install prebuilt rules
+And user should NOT see a number of rules available to install
+```
+
+#### **Scenario: Empty screen is shown when all prebuilt rules are installed**
+
+**Coverage**: 1 e2e test with mock rules.
+
+```Gherkin
+Given all the available prebuilt rules are installed in Kibana
+When user opens the Add Rules page
+Then user should see a message indicating that all available rules have been installed
+And user should see a CTA that leads to the Rule Management page
+```
+
+### Rule installation workflow: filtering, sorting, pagination
+
+TODO: add scenarios
```

Comment:
TODO: @banderror sync with @jpdjere

Reply:
Synced. The TODO is left in the file -- we can work on adding these scenarios and tests for them in 8.11.